File: workspace/juha/Controller_EP_Integration/src/org/openremote/controller/model/event/Switch.java
/*
 * OpenRemote, the Home of the Digital Home.
 * Copyright 2008-2011, OpenRemote Inc.
 *
 * See the contributors.txt file in the distribution for a
 * full listing of individual contributors.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
package org.openremote.controller.model.event;

import org.openremote.controller.protocol.Event;

/**
 * TODO
 *
 * @author <a href="mailto:juha@openremote.org">Juha Lindfors</a>
 */
public class Switch extends Event<Switch.State>
{
  public enum State { ON, OFF }

  private State eventState;

  public Switch(int sourceID, String sourceName, State state)
  {
    super(sourceID, sourceName);

    this.eventState = state;
  }

  public Switch.State getValue()
  {
    return eventState;
  }

  public String serialize()
  {
    return eventState.toString().toLowerCase();
  }
}

File: workspace/ebariaux/iPad_Console_2_0_0_Betas-SIPIntegration/Classes/Switch.m
/* OpenRemote, the Home of the Digital Home.
 * Copyright 2008-2009, OpenRemote Inc.
 * 
 * See the contributors.txt file in the distribution for a
 * full listing of individual contributors.
 * 
 * This is free software; you can redistribute it and/or modify it
 * under the terms of the GNU General Public License as
 * published by the Free Software Foundation; either version 3.0 of
 * the License, or (at your option) any later version.
 * 
 * This software is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * 
 * You should have received a copy of the GNU General Public
 * License along with this software; if not, write to the Free
 * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
 * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
 */

#import "Switch.h"
#import "Image.h"
#import "SensorState.h"

@implementation Switch

@synthesize onImage, offImage;


//get element name, must be overriden in subclass
- (NSString *) elementName {
	return SWITCH;
}


#pragma mark Delegate methods of NSXMLParser  


- (id)initWithXMLParser:(NSXMLParser *)parser elementName:(NSString *)elementName attributes:(NSDictionary *)attributeDict parentDelegate:(NSObject<NSXMLParserDelegate> *)parent {
	if (self = [super init]) {
		componentId = [[attributeDict objectForKey:ID] intValue];
		
		xmlParserParentDelegate = [parent retain];
		[parser setDelegate:self];
	}
	return self;
}

/**
 * Fill the switch on/off state images .
 */
- (void)parser:(NSXMLParser *)parser didEndElement:(NSString *)elementName namespaceURI:(NSString *)namespaceURI qualifiedName:(NSString *)qName {
	if ([elementName isEqualToString:[self elementName]]) {	
		
		for (SensorState *state in sensor.states) {
			Image *img = [[Image alloc] init];
			img.src = [state.value copy];
			if ([[state.name lowercaseString] isEqualToString:ON]) {
				onImage = img;
			} else if ([[state.name lowercaseString] isEqualToString:OFF]) {
				offImage = img;
			}
			
		}
				
 		[parser setDelegate:xmlParserParentDelegate];
		[xmlParserParentDelegate release];
		xmlParserParentDelegate = nil;
	}
}

- (void)dealloc {
	[onImage release];
	[offImage release];
	
	[super dealloc];
}


@end
